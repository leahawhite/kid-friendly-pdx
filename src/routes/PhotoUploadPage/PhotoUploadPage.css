.container {
  display: flex;
  flex-direction: column;
  height: 75vh;
  justify-content: space-around;
}

.upload-btn {
  display: flex;
  flex-direction: column;
  text-align: center;
  border: 1px solid #00b8e6;
  background-color:  #00b8e6;
  color: white;
  font-weight: bold;
  font-size: 18px;
  padding: 15px 30px;
  border: 1px solid #ddd;
  -moz-border-radius: 8px;
  -webkit-border-radius: 8px;
  -o-border-radius: 8px 12px;
  -ms-border-radius: 8px;
  border-radius: 8px;
  margin: 0 auto;
}

.upload-btn span {
  align-self: center;
}

.buttons {
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex-wrap: wrap;
  margin: 20px;
  color: #00b8e6;
}

.button {
  margin: 0 5vw
}

.cancel-btn {
  border: 1px solid #00b8e6;
  background-color: #00b8e6;
  color: white;
  font-weight: bold;
  font-size: 18px;
  padding: 8px;
  -moz-border-radius: 8px;
  -webkit-border-radius: 8px;
  -o-border-radius: 8px 12px;
  -ms-border-radius: 8px;
  border-radius: 8px;
  margin: 0 auto;
  align-self: center;
  margin-bottom: 60px;
}

.upload-header {
  font-size: 22px;
  margin-top: 40px;
}

.images {
  display: flex;
  justify-content: center;
}

.delete {
  position: relative;
  top: 16px;
  right: 15px
}

.button svg, .delete svg {
  filter: drop-shadow(3px 3px 3px #999);
  transition: 0.3s;
  transform: rotate(-0.2deg)
}

.button svg:hover, .delete svg:hover {
  filter: drop-shadow(4px 4px 4px #999);
  transform: rotate(-0.6deg);
  cursor: pointer
}

.delete svg {
  filter: drop-shadow(1px 1px 1px #999)
}

.delete svg:hover {
  filter: drop-shadow(1.5px 1.5px 1.5px)
}

input[type=file] { 
  visibility: hidden;
  position: absolute
}

.uploaded-images {
  max-height: 420px;
  max-width: 420px;
  padding: 10px;
  border: 1px dashed #999;
}

/* Loading */
.loading-wrapper, .loading {
  display: flex;
  align-items: center;
  justify-content: center; 
  flex-direction: column;
}

.loading {
  width: 200px;
  height: 200px;
}

.loading .background {
  border-radius: 50%;
  background: #6762a6;
  border: 3px solid #c9c3e6;
  box-shadow: 1px 2px 2px rgba(0, 0, 0, 0.25);
  width: 200px;  
  height: 200px;
  box-sizing: border-box;
  animation: pulse-colors 2s infinite alternate linear;
}

.loading i {
  margin: 25px 5px 5px 55px;
  float: left;
  font-size: 10em !important;
  color: #fff;
  animation: pulse-icon 4s infinite alternate linear;
}

#notification-wrapper {
  position: absolute;
}


@keyframes pulse-colors {
  from {
    background: #c9c3e6;
    border: 3px solid #a29ccc;
  }
  to {
    background: #6762a6;
    border: 3px solid #c9c3e6;
  }
}

/* Animations */
.fadein {
  animation: fadein 2s;
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

.spinner {
  animation: spin 1.2s infinite linear;
}

.spinner svg {
  background: #f2f2f2;
  border-radius: 50%
}

@-webkit-keyframes spin {
  0%  {
    -webkit-transform: rotate(0deg)
  }
  100% {
    -webkit-transform: rotate(360deg)
  }   
}

.rolling {
  position: absolute;
  animation: move 6s ease-out;
  filter: drop-shadow(4px 4px 4px #999);
}

@-webkit-keyframes move {
  0%  {
    left: -5%;
  }
  100% {
    left: 105%;
  }   
}

@keyframes pulse-icon {
  from {
    text-shadow: 1px 2px 2px rgba(0, 0, 0, 0.25);
  }
  to {
    text-shadow: 2px 4px 4px rgba(0, 0, 0, 0.55);
  }
}

