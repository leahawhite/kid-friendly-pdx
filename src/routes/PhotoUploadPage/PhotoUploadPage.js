import React, { Component } from 'react'
import Spinner from '../../components/Spinner/Spinner'
import Images from '../../components/Images/Images'
import UploadButton from '../../components/UploadButton/UploadButton'
import Button from '../../components/Button/Button'
import TokenService from '../../services/token-service'
import config from '../../config'
import './PhotoUploadPage.css'

export default class UploadPhotosPage extends Component {
  state = {
    uploading: false,
    selectedFile: null,
    images: [],
    error: null,
    captions: []
  }

  handleCancel = () => {
    const { history } = this.props
    history.goBack()
  }

  onChange = e => {
    this.setState({
      selectedFile: e.target.files
    })
  }

  onSubmit = e => {
    e.preventDefault()
    this.setState({ uploading: true })
    // this might be screwing up multer
    // const files = Array.from(e.target.files)
    
    const formData = new FormData()
    for (let i = 0; i < this.state.selectedFile.length; i++) {
      formData.append('file', this.state.selectedFile[i])
    }
    
    fetch(`${config.API_ENDPOINT}/images/upload`, {
      method: 'POST',
      body: formData
    })
    .then(res =>
      (!res.ok)
        ? res.json().then(e => Promise.reject(e))
        : res.json()
    )
    .then(images => {
      console.log('images', images)
      this.setState({ 
        uploading: false,
        images
      })
    })
    .catch(res => {
      this.setState({ error: res.error })
    })
  }
    
  filter = id => {
    return this.state.images.filter(image => image.public_id !== id)
  }

  removeImage = id => {
    this.setState({
      images: this.state.images.filter(image => image.public_id !== id)
    })
  }

  addCaption = text => {
    this.setState({
      captions: [...this.state.captions, text]
    })
  }

  onError = id => {
    this.setState({ images: this.filter(id) })
  }

  /*submitImagestoDB = (e) => {
    e.preventDefault()
    const { images } = this.state
    const { place } = this.props.location.state
    // need to deconstruct the images array to get properties
    images.map(image => {
      const newImage = {
      id: image.public_id,
      src: image.secure_url,
      place_id: place.id,
      // caption: how to get this from array in this.state? 
      // user_id: TBD
      }
      return newImage
    })
    fetch(`${config.API_URL}/images`, {
      method: 'POST',
      headers: {
        "Authorization": `basic ${TokenService.getAuthToken()}`,
        "Content-Type": "application/json"
      },
      body: JSON.stringify(newImage)
      })
      .then(res =>
        (!res.ok)
          ? res.json().then(e => Promise.reject(e))
          : res.json()
      )
        .then(data => {
          e.target.text.value = ''
        })
        .catch(res => {
          this.setState({ error: res.error })
        })
        this.setState({
          fireRedirect: true,
        })
    }*/
  
  render() {
    const { uploading, images } = this.state

    const content = () => {
      switch(true) {
        case uploading:
          return <Spinner />
        case images.length > 0:
          return ( 
                <>
                  <Images 
                    images={images} 
                    removeImage={this.removeImage} 
                    addCaption={this.addCaption}
                    onError={this.onError}
                  />
                  <div className="btn-container">
                    <Button btnType="submit" btnText="Finished" />
                  </div>
                </>
                 )
        default:
          return (
            <>
            <UploadButton iconClass="icon" uploadSpan="Browse Files" onChange={this.onChange} />
            <Button btnType="submit" btnText="Submit"/>
            </>
          ) 
      }
    }
    const hasError = this.state.error && <div className="error">{this.state.error}</div>

    return (
      <div className='upload-container'>
        <h2 className="upload-header">Upload Photos</h2>
        <form className="upload-form" onSubmit={this.onSubmit} action="/api/images/upload" method="post" encType="multipart/form-data">
          <div className='buttons'>
            {content()}
          </div>
          {hasError}
          <div className="cancel-container">
            <Button btnText="Cancel" btnType="button" onClick={this.handleCancel} />
          </div>
        </form>
      </div>
    )
  }
}