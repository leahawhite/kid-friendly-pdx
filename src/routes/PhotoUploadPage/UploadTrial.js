import React, { Component } from 'react'
import Spinner from '../../components/Spinner/Spinner'
import Images from '../../components/Images/Images'
import UploadButton from '../../components/UploadButton/UploadButton'
import Button from '../../components/Button/Button'
import config from '../../config'
import './PhotoUploadPage.css'

export default class UploadTrial extends Component {
  state = {
    uploading: false,
    selectedFiles: null,
    images: []
  }

  onSubmit = e => {
    this.setState({ uploading: true })
    // this might be screwing up multer
    // const files = Array.from(e.target.files)
    
    const formData = new FormData()
    for (let x = 0; x < this.state.selectedFiles.length; x++) {
      formData.append('myImages', this.state.selectedFiles[x])
    }
    
    fetch(`${config.API_ENDPOINT}/images/upload`, {
      method: 'POST',
      body: formData
    })
    .then(res =>
      (!res.ok)
        ? res.json().then(e => Promise.reject(e))
        : res.json()
    )
    .then(images => {
      this.setState({ 
        uploading: false,
        images
      })
    })
    .catch(res => {
      this.setState({ error: res.error })
    })
  }
  /*onSubmit = e => {
    e.preventDefault()
    const formData = new FormData()
    console.log('formdata', Array.from(formData.entries()))
    fetch(`${config.API_ENDPOINT}/images/upload`, {
      method: 'POST',
      body: formData
    })
    .then(res => res.json())
    .then(images => {
      this.setState({
        uploading: false, 
        images: images
      })
    })
    .catch(error => console.error('Error:', error))
  }*/

  onChange = e => {
    this.setState({
      selectedFiles: e.target.files
    });
    
  }

  render() {
    const { images } = this.state
    const results = images && images.map((image, idx) => (
      <div key={idx} className="reviewImage">
        <img className="uploaded-images" src={image.secure_url} alt={image.id} />
      </div>
  ))
    return (
      <>
      <form action='/api/images/upload' method="post" encType="multipart/form-data" onSubmit={this.onSubmit}>
        <label htmlFor="file">Upload file</label>
        <input type='file' name='myImages' id="myImages" multiple onChange={this.onChange}/>
        <button type="submit">Submit</button>
      </form>
      {results}
      </>
    )
  }
}