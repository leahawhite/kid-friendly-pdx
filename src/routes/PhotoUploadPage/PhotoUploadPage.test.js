import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PhotoUploadPage from './PhotoUploadPage'

describe('PhotoUploadPage component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <PhotoUploadPage />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<PhotoUploadPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
