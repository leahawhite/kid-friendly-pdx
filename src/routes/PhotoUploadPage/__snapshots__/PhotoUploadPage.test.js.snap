// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoUploadPage component renders UI as expected 1`] = `
<div
  className="container"
>
  <h2
    className="upload-header"
  >
    Upload Photos
  </h2>
  <div
    className="buttons"
  >
    <_default />
  </div>
  <div
    className="cancel-container"
  >
    <button
      className="cancel-btn"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;
